src/components/hint/Hint.driver.new.ts

```diff
@@ -0,0 +1,90 @@
+import {HintProps} from './index';
+import {TouchableOpacityProps} from '../touchableOpacity';
+import {
+  useComponentDriver,
+  ComponentProps,
+  usePressableDriver,
+  ViewDriver,
+  ImageDriver,
+  ModalDriver,
+  TextDriver
+} from '../../testkit';
+import {ViewStyle, StyleSheet} from 'react-native';
+
+export const HintDriver = (props: ComponentProps) => {
+  const driver = usePressableDriver<HintProps>(useComponentDriver(props));
+
+  const modalDriver = ModalDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}.modal`
+  });
+
+  const contentDriver = ViewDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}.message`
+  });
+
+  const textDriver = TextDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}.text`
+  });
+
+  const hintPressableDriver = usePressableDriver<TouchableOpacityProps>(useComponentDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}`
+  }));
+
+  const overlayDriver = usePressableDriver<TouchableOpacityProps>(useComponentDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}.overlay`
+  }));
+
+  const iconDriver = ImageDriver({
+    renderTree: props.renderTree,
+    testID: `${props.testID}.icon`
+  });
+
+  const getModal = () => {
+    return modalDriver;
+  };
```

Comment:
I think this is an internal implementation detail, you probably want to externalize `isVisible` instead